frontend/src/views/main/project/ProjectsHome.vue

```diff
@@ -92,90 +92,96 @@
   </div>
 </template>
 
-<script lang="ts">
-import {Component, Vue, Watch} from 'vue-property-decorator';
-import {ValidationObserver} from 'vee-validate'
-import {readAllProjects, readHasAdminAccess, readUserProfile} from '@/store/main/getters';
-import {Role} from '@/enums';
-import {dispatchCreateProject, dispatchGetProjects} from '@/store/main/actions';
-import {ProjectPostDTO} from '@/generated-sources';
+<script setup lang="ts">
+import {computed, onMounted, ref, watch} from "vue";
+import {ValidationObserver} from "vee-validate";
+import {dispatchCreateProject, dispatchGetProjects} from "@/store/main/actions";
+import {readAllProjects, readHasAdminAccess, readUserProfile} from "@/store/main/getters";
+import {Role} from "@/enums";
+import {ProjectPostDTO} from "@/generated-sources";
+import {toSlug} from "@/utils";
+import store from "@/store";
+import {useRoute, useRouter} from "vue-router/composables";
 import moment from "moment";
-import {toSlug} from '@/utils';
 
-@Component
-export default class ProjectsHome extends Vue {
+const route = useRoute();
+const router = useRouter();
 
-  private openCreateDialog = false; // toggle for edit or create dialog
-  private newProjectName = "";
-  private newProjectKey = "";
-  private newProjectDesc = "";
-  private creatorFilter = 0;
-  private projectCreateError = "";
+const openCreateDialog = ref<boolean>(false); // toggle for edit or create dialog
+const newProjectName = ref<string>("");
+const newProjectKey = ref<string>("");
+const newProjectDesc = ref<string>("");
+const creatorFilter = ref<number>(0);
+const projectCreateError = ref<string>("");
 
-  $refs!: {
-    dialogForm: InstanceType<typeof ValidationObserver>;
-  };
 
-  public async mounted() {
-    const f = this.$route.query.f ? this.$route.query.f[0] || "" : ""
-    this.creatorFilter = parseInt(f) || 0;
-    await this.loadProjects();
-  }
+// template ref
+const dialogForm = ref<ValidationObserver | null>(null);
 
-  private async loadProjects() {
-    await dispatchGetProjects(this.$store);
-  }
+onMounted(async () => {
+  const f = route.query.f ? route.query.f[0] || '' : '';
+  creatorFilter.value = parseInt(f) || 0;
+  await loadProjects();
+})
 
-  get userProfile() {
-    return readUserProfile(this.$store);
-  }
+// computed
+const userProfile = computed(() => {
+  return readUserProfile(store);
```

Comment:
`readUserProfile` returns `AdminUserDTO | null` so `userProfile` can be null too.
So either you declare computed as `computed<AdminUserDTO | null>...` or handle the null inside the computed method.
If we do declare computed as nullable, then in the template we'd have to add `?` everywhere where 
In the current context null user profile doesn't make sense. If you reached projects home page you there should be user profile already since the user is logged in. So I suggest we do something like this:

```

// computed
const userProfile = computed<AdminUserDTO>(() => {
  const userProfile = readUserProfile(store);
  if (userProfile === null){
    throw new Error("User is not defined");
  }
  return userProfile;
});
```